### PR TITLE
Ekitabu/feature/dev

### PR DESCRIPTION
**Pull request by brettins**
Analytics working again. 

openBook
closeBook
openApp
closeApp

events should all be firing.

### CODE REVIEW

**Comment by mutterback** [comment]
Thank you!

**Comment by kevinmainairungu** [comment]
@brettins getting this error when I run `npm start`
Screen Shot 2020-10-27 at 6 42 41 PM
I'm also getting a node-gyp error using mvsc 2017 build tools. I've tried deleting my node modules and running `npm install` but that does not successfully complete because of this. I've also tried to update node-gyp to v.7. That also hasn't solved it.My global version of Node is 12.9.0 npm is 6.14.8.

**Comment by brettins** [comment]
Hey @kevinmainairungu, the SQLITE_CANTOPEN should be fixed with the latest commit I've added. The other solution to it was removing your roaming folder, but I figured it would be easier to adjust it this way.

I'm not sure about the `node-gyp` error, weren't you having that before and solved it by updating gyp?

**Comment by mutterback** [comment]
@kevinmainairungu are you able to review this Tuesday?

**Comment by kevinmainairungu** [comment]
> @kevinmainairungu are you able to review this Tuesday?

Yes, @mutterback, we can go over this today.

**Comment by kevinmainairungu** [comment]
> Hey @kevinmainairungu, the SQLITE_CANTOPEN should be fixed with the latest commit I've added. The other solution to it was removing your roaming folder, but I figured it would be easier to adjust it this way.
> 
> I'm not sure about the `node-gyp` error, weren't you had that before and solved it by updating gyp?

@brettins I managed to solve the issue on node-gyp. It was an error with my visual studio build tools. After I repaired the build tools, the error was no longer there. Also, the SQLITE_CANTOPEN error is solved.

**Comment by mutterback** [comment]
@kevinmainairungu does your comment mean you've now reviewed this and approved?

**Comment by brettins** [comment]
@kevinmainairungu please also note that the responses you are giving so far are QA work, they are not code review responses.

**Comment by kevinmainairungu** [comment]
> @kevinmainairungu does your comment mean you've now reviewed this and approved?

Not yet @mutterback, I still have to go over the code that brett wrote, the files that changed and the comments he left out on the functionality. I'm still doing so will be done soon.

File: src/main/di.ts [hunk 1]
```diff
@@ -57,13 +57,15 @@ import { RootState } from "./redux/states";
 import { OpdsService } from "./services/opds";
 
 //TODO - get rid of these when moving db generation to another file 
-import * as uuid from "uuid";
+import { v4 as uuidv4 } from "uuid";
```
Comment (kevinmainairungu) [inline src/main/di.ts:60]: npm UUID version 4

Comment (kevinmainairungu) [inline src/main/di.ts:60]: changed npm UUID to version 4

File: src/main/di.ts [hunk 2]
```diff
@@ -57,13 +57,15 @@ import { RootState } from "./redux/states";
...
 // Logger
 const debug = debug_("readium-desktop:main:di");
 
 export const CONFIGREPOSITORY_REDUX_PERSISTENCE = "CONFIGREPOSITORY_REDUX_PERSISTENCE";
 const capitalizedAppName = _APP_NAME.charAt(0).toUpperCase() + _APP_NAME.substring(1);
```
Comment (kevinmainairungu) [inline src/main/di.ts:65]: @brettins is this line responsible for cabitalizing the E in eKitabu?

**Review by kevinmainairungu — APPROVED** [review]
@brettins I checked the new PouchDBSqlite instance and now we are using UUIDv4.

**Comment by brettins** [comment]
@kevinmainairungu yes it likely is the line causing the uppercase E.

**Pull request merged**